catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/EntityRepository.java

```diff
@@ -588,6 +585,198 @@ private void recoverDeletedRelationships(T original) {
     }
   }
 
+  /** Decorator class for Entity. */
+  public class EntityHandler {
+    private final Include isDeleted;
+    protected final EntityInterface<T> entityInterface;
+    private final T entity;
+
+    public EntityHandler(T entity) {
+      this.entityInterface = getEntityInterface(entity);
+      this.entity = entity;
+      this.isDeleted = entityInterface.isDeleted() ? DELETED : Include.NON_DELETED;
+    }
+
+    EntityReference toEntityReference() {
+      return entityInterface.getEntityReference();
+    }
+
+    /**
+     * Validate the owner if not null before creating this resource. Owner must exist, not being deleted, and must be
+     * either User or Team.
+     *
+     * @return
+     */
+    public EntityReference validateOwnerOrNull() {
+      EntityReference entityReference = validateFieldOrNull("owner");
+      if (entityReference == null) {
+        return null;
+      } else if (!List.of(Entity.USER, Entity.TEAM).contains(entityReference.getType())) {
+        throw new IllegalArgumentException(String.format("Invalid type %s", entityReference.getType()));
+      }
+      return entityReference;
+    }
+
+    /**
+     * Validate a field containing a EntityReference if not null before creating this resource. The target entity must
+     * exist and not deleted.
+     *
+     * @param fieldName
+     * @return
+     */
+    @SneakyThrows
+    public EntityReference validateFieldOrNull(String fieldName) {
+      Method method = entity.getClass().getMethod("get" + StringUtils.capitalize(fieldName));
+      EntityReference entityReference = (EntityReference) method.invoke(entity);
+      if (entityReference == null) {
+        return null;
+      }
+      // this could be changed to Include.NON_DELETED because we validate only when creating entities linked to
+      // non-deleted entities.
+      Object entity = Entity.getEntity(entityReference, Fields.EMPTY_FIELDS, isDeleted);
+      return h(entity).toEntityReference();
+    }
+
+    /**
+     * When listing and getting resources, we need to get entities using the relationships. If the relationship exists
+     * then the entity must exist.
+     *
+     * @return
+     */
+    @SneakyThrows
+    public EntityReference getOwnerOrNull() {
+      List<EntityReference> refs =
+          daoCollection
+              .relationshipDAO()
+              .findFrom(
+                  entityInterface.getId().toString(), entityName, Relationship.OWNS.ordinal(), toBoolean(isDeleted));
+      if (refs.isEmpty()) {
+        return null;
+      } else if (refs.size() > 1) {
+        LOG.warn("Possible database issues - multiple relationships found for entity {}", entityInterface.getId());
```

Comment:
Can we also add entity type in the log, so that it helps with debugging if needed?

Reasoning:
- With the type explicitly mentioned in the log line, it is easy for us to go directly to the corresponding table in the database

Reply:
Done.